src/Timeline.cpp

```diff
@@ -379,8 +379,8 @@ void Timeline::add_layer(std::shared_ptr<Frame> new_frame, Clip* source_clip, in
 
 	}
 
-	// Skip out if only an audio frame
-	if (!source_clip->Waveform() && !source_clip->Reader()->info.has_video)
+	// Skip out if video was disabled or only an audio frame (no visualisation in use)
+	if (source_clip->has_video.GetInt(clip_frame_number) == 0 || !source_clip->Waveform() && !source_clip->Reader()->info.has_video)
```

Comment:
Are we missing some parenthesis?
`if (source_clip->has_video.GetInt(clip_frame_number) == 0 || (!source_clip->Waveform() && !source_clip->Reader()->info.has_video))`

Reply:
Is it required change? If indentation needed I can make it look like this:
```
if (source_clip->has_video.GetInt(clip_frame_number) == 0 ||
    !source_clip->Waveform() && !source_clip->Reader()->info.has_video)
```
Maybe it will be easier to read than additional pair of parentheses.